misc-scripts/xref_mapping/XrefParser/RefSeqCoordinateParser.pm

```diff
@@ -23,405 +23,300 @@ use strict;
+  # Add link to EntrezGene IDs where available
+  my $entrez_ids = $self->get_valid_codes("EntrezGene", $species_id, $dbi);
+
   my $entrez_source_id = $self->get_source_id_for_source_name('EntrezGene', undef, $dbi);
-  my $add_dependent_xref_sth = $dbi->prepare("INSERT INTO dependent_xref  (master_xref_id,dependent_xref_id, linkage_source_id) VALUES (?,?, $entrez_source_id)");
 
   my $sa = $core_dba->get_SliceAdaptor();
   my $sa_of = $otherf_dba->get_SliceAdaptor();
   my $chromosomes_of = $sa_of->fetch_all('toplevel', undef, 1);
 
-# Fetch analysis object for refseq
+  # Fetch analysis object for refseq
   my $aa_of = $otherf_dba->get_AnalysisAdaptor();
-  my $logic_name;
-  foreach my $ana(@{ $aa_of->fetch_all() }) {
-    if ($ana->logic_name =~ /refseq_import/) {
-      $logic_name = $ana->logic_name;
-    }
-  }
-## Not all species have refseq_import data, skip if not found
-  if (!defined $logic_name) {
-    print STDERR "No data found for RefSeq_import, skipping import\n";;
-    return;
+
+  # Not all species have refseq_import data, skip if not found
+  if (!defined $aa_of->fetch_by_logic_name('refseq_import')->logic_name) {
+    croak "No data found for RefSeq_import, skipping import\n";
   }
 
-  foreach my $chromosome_of (@$chromosomes_of) {
+  # start main loop
+  foreach my $chromosome_of (@{$chromosomes_of}) {
     my $chr_name = $chromosome_of->seq_region_name();
-    my $genes_of = $chromosome_of->get_all_Genes($logic_name, undef, 1);
+    my $genes_of = $chromosome_of->get_all_Genes('refseq_import', undef, 1);
 
-    while (my $gene_of = shift @$genes_of) {
+    foreach my $gene_of (@{$genes_of}) {
       my $transcripts_of = $gene_of->get_all_Transcripts();
 
-# Create a range registry for all the exons of the refseq transcript
-      foreach my $transcript_of (sort { $a->start() <=> $b->start() } @$transcripts_of) {
-	my ($id, $tl_id);
-	# We're moving to RefSeq accessions being stored as xrefs rather than
-	# stable ids. But we also need to maintain backwards compatbility.
-	# If it's the new kind, where there's a display_xref use that,
-	# otherwise fall back to using the stable_id. But also check if we
-	# have neither, then skip the record.
-	if (defined $transcript_of->display_xref ) {
-	  $id = $transcript_of->display_xref->display_id;
-	} elsif (defined $transcript_of->stable_id) {
-	  $id = $transcript_of->stable_id;
-	} else {
-	  # Skip non conventional accessions
-	  next;
-	}
-        if ($id !~ /^[NXMR]{2}_[0-9]+/)  { next; }
-        my %transcript_result;
-        my %tl_transcript_result;
-        if (!defined $id) { next; }
+      # Create a range registry for all the exons of the refseq transcript
+      foreach my $transcript_of (sort { $a->start <=> $b->start } @{$transcripts_of}) {
+        my $id;
+        # We're moving to RefSeq accessions being stored as xrefs rather than
+        # stable ids. But we also need to maintain backwards compatbility.
```

Comment:
Sentences should not generally begin with 'But'. In general this comment suggests you understand what is going on, but it could be made clearer to another reader. What 'new kind' do you refer to here?